src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1959,11 +1959,17 @@ internal static bool AccessingAutopropertyFromConstructor(BoundExpression receiv
             return (object)sourceProperty != null &&
                     sourceProperty.IsAutoProperty &&
                     sourceProperty.ContainingType == fromMember.ContainingType &&
-                    ((MethodSymbol)fromMember).MethodKind == (propertyIsStatic ? MethodKind.StaticConstructor
-                                                                                : MethodKind.Constructor) &&
-                   (propertyIsStatic || receiver.Kind == BoundKind.ThisReference);
+                    IsConstructorOrField(fromMember, isStatic: propertyIsStatic) &&
+                    (propertyIsStatic || receiver.Kind == BoundKind.ThisReference);
         }
 
+        private static bool IsConstructorOrField(Symbol member, bool isStatic)
```

Comment:
This function doesn't quite do what it says -- it doesn't ask whether or not the member is a constructor, it asks if it's a constructor with the same receiver type as given in isStatic. I would consider changing the name of the method or the parameter name to make it clear that the parameter is meant to indicate the expected receiver of the member.